### PR TITLE
Update structure of table exchange rate, Complete API for exchange ra…

### PR DESCRIPTION
**Pull request by lomamech**
- Update structure of table exchange rate, 
- Complete API for exchange rate, 
- and completion of test integration

@jniles @DedrickEnc

### CODE REVIEW

**Comment by jniles** [comment]
@lomamech , did you run your integration tests?

**Comment by lomamech** [comment]
@jniles  I have Changing the attribute enterprise_currency_id and foreign_currency_id in all the file on server and on client and i have also updating the test api for patient because this test return an error message,

File: server/controllers/finance/exchange.js [hunk 1]
```diff
@@ -42,13 +42,62 @@ exports.create = function create(req, res, next) {
...
+
+  var sql =
+    'DELETE FROM exchange_rate WHERE id = ?;';
+
+  db.exec(sql, [req.params.id])
+  .then(function (row) {
+
+    // if nothing happened, let the client know via a 404 error
+    if (row.affectedRows === 0) {
+      return res.status(404).send();
```
Comment (jniles) [inline server/controllers/finance/exchange.js]: You should use the response error codes defined  in `codes.js`.  See the /conventions or /cash API to see how it is used.

File: server/test/api/patients.js [hunk 2]
```diff
@@ -56,8 +56,7 @@ describe('The /patients API', function () {
   var mockMissingRequest = {
     finance : {
       debitor_group_uuid : '4de0fe47-177f-4d30-b95f-cff8166400b4'
-    },
-    medical : missingParamsPatient
```
Comment (jniles) [inline server/test/api/patients.js:60]: Why were these removed?  They don't seem to be related to exchange rates.

Comment (lomamech) [inline server/test/api/patients.js:60]: @jniles I changed because all the walking test integration with the exeption of " POST / patients will reject a patient with missing information"

File: server/test/api/exchange.js [hunk 3]
```diff
@@ -51,4 +54,32 @@ describe('The /exchange API endpoint', function () {
...
+      .send({ rate : 925 })
+      .then(function (res) {
+        expect(res).to.have.status(200);
+        expect(res.body).to.have.keys(RATE_KEY);
+        expect(res.body.rate).to.not.equal(RATE.rate);
+      })
+      .catch(helpers.handler);
+  });
+
+  it('DELETE /exchange/:id will send back a 404 if the exchage rate does not exist', function () {
```
Comment (jniles) [inline server/test/api/exchange.js:80]: The API should send back an error code to translate in case of an error.  Please check that it exists.

**Comment by jniles** [comment]
All tests pass!  That's excellent.

Could you include a test for `PUT /exchange/:undefinedId`?  It should return a 404 with a translatable error code.

This is a very good PR.   I've made a few comments, most of them just standardization with other modules/APIs.

@IMA-WorldHealth/local-contributors , please look at the changes in this PR.  They will profoundly affect how we use exchange rates in our modules.

File: server/test/api/patients.js (hunk 2 above)
Comment (jniles) [inline server/test/api/patients.js:60]: If the test does not pass, it means the test found a bug.  Why did changes to exchange rate break the patients API?  All tests pass on master.

**Comment by lomamech** [comment]
@jniles Here is the modification and test updated an exchange rate that does not exist

**Comment by jniles** [comment]
Much better!  Everything LGTM.
